### DISCUSSION TITLE
Viewport slows down when locking Camera to View

### DISCUSSION THREAD

**Discussion by BFDWPH** [bnpr/Malt#376] [Feedback & Feature Requests]
Hi,

I have a problem when using Malt during preview renders in the viewport - it updates great until I lock the camera to the viewport, then the update gets laggy. Unlock the view again while still in Malt render preview, no problem. I've searched other reports of viewport update problems, but didn't see this issue...

Thanks,

William

**Comment by BFDWPH** [comment]
An update on above: the view slows when I move the actual camera, object once Malt is used as the preview renderer. The camera doesn't need to be lcoked to the view.

**Reply by pragma37** [reply]
Does it slow down when you move any other object?

Any update in the scene triggers a reload of all objects properties (object matrix, render settings...) at the moment.
If your scene has many objects or particles this is expected, at least until we optimize it.

**Reply by BFDWPH** [reply]
Hi, thanks for the reply. 

Screen updates are fine until until I switch to camera view with camera lock is enabled. Then, it slows down right away.

It doesn't matter if I move the camera while locked in view, or if I move an object while camera is lock in view. Once I disable camera viewlock, the scene updates smoothly again. I've tried this  with small and medium sized scenes.

I found I can mostly work around it by not using camera lock, and just aligning the camera to the viewport when I have a shot I liked, 
Other than that, after a couple of weeks use, Malt is great :)
